### ISSUE TITLE
Deceleration of fit by 10 times, after tf.keras.models.load_model('model.h5')

### ISSUE THREAD

**Issue by nikita1andreev** [tensorflow/tensorflow#39459]
I had the same problem as:
https://github.com/tensorflow/tensorflow/issues/34767

After installing tf-nightly-gpu" and checked scenario 3: model = Model_Functional_API() + model.save('saved_model.h5'), crashes are gone.

But, fit speed after model loading from file, became very slow.
When i create model like "modelC = Model(...)", fit from cold start was about 4-5 min, on next fit it was 3 min.

Fit on model loaded from file show ETA above 30 min.
Recompiliing after load has no effect.

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):

if os.path.isfile("model.h5"):
    modelC = tf.keras.models.load_model('model.h5')#    
else: 
    if1 = Input(shape=(tr1.shape[1], tr1.shape[2]), name='input1')
    if2 = Input(shape=(tr2.shape[1], tr2.shape[2]), name='input2')
    if3 = Input(shape=(tr3.shape[1], tr3.shape[2]), name='input3')
    if4 = Input(shape=(tr4.shape[1], tr4.shape[2]), name='input4')
    x1 = LSTM(tr1.shape[2], batch_input_shape=(batch_size,timesteps,tr1.shape[2]), return_sequences=True)(if1)
    x2 = LSTM(tr2.shape[2], batch_input_shape=(batch_size,timesteps,tr2.shape[2]), return_sequences=True)(if2)
    x3 = LSTM(tr3.shape[2], batch_input_shape=(batch_size,timesteps,tr3.shape[2]), return_sequences=True)(if3)
    x4 = LSTM(tr4.shape[2], batch_input_shape=(batch_size,timesteps,tr4.shape[2]), return_sequences=True)(if4)
    combined = concatenate([x1, x2, x3, x4])
    x = LSTM(512, return_sequences=True)(combined)
    x = TimeDistributed(Dense(tst.shape[2]*4, activation="softmax"))(x)
    out = Dense(tst.shape[2], activation="softmax")(x)
    modelC = Model(inputs=[if1,if2,if3,if4], outputs=out)
    modelC.save("model.h5")
modelC.compile(optimizer=tf.keras.optimizers.Adam(lr=1e-4)
,loss=tf.keras.losses.CategoricalCrossentropy(from_logits=False,label_smoothing=0, reduction=tf.keras.losses.Reduction.AUTO )
,metrics=[tf.keras.metrics.CategoricalAccuracy()])

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Fedora 27 (Workstation Edition)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):

tf-estimator-nightly   2.3.0.dev2020051201
tf-nightly             2.2.0.dev20200508
tf-nightly-gpu         2.2.0.dev20200508
tensorboard            2.1.1
tensorboard-plugin-wit 1.6.0.post3
tensorflow             2.1.0
tensorflow-estimator   2.1.0
tensorflow-gpu         2.1.0
tensorrt               7.0.0.11

- TensorFlow version (use command below):
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
unknown 2.1.0

- Python version:
Python 3.7.7

- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
libcudnn7.x86_64                       7.6.5.33-1.cuda10.2             @nvidia-machine-learning
libcudnn7-devel.x86_64                 7.6.5.33-1.cuda10.2             nvidia-machine-learning

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.64.00    Driver Version: 440.64.00    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 2060    Off  | 00000000:04:00.0 Off |                  N/A |
| 46%   53C    P2    38W / 160W |   4392MiB /  5934MiB |     38%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1271      G   /usr/libexec/Xorg                              9MiB |
|    0      1507      G   /usr/bin/gnome-shell                           8MiB |
|    0      7195      C   /root/anaconda3/bin/python                  4361MiB |
+-----------------------------------------------------------------------------+

- GPU model and memory:
RTX 2060 
72GB DDR4

**Describe the current behavior**
30 min on fit after loading model

**Describe the expected behavior**
3 min of fit after loading model, like when i contsruct model

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by Saduf2019** [comment]
@nikita1andreev 
I ran the code shared and see indentation errors, can you please provide with simple stand alone code such that we can replicate the issue faced, if possible please share a colab gist to analyse the error.

**Comment by nikita1andreev** [comment]
i am afraid, i have'nt example on colab, because they kick me out when i exceeded limit :).
so i bought RTX, and use model at home. 
data size about 20Gb, and it confidential.
but you cat generate random.
tr1-3  shape: (1854, 3600, 401)
tr4.shape: (1854, 3600, 7)
tst.shape: (1854, 3600, 3)

**Comment by Saduf2019** [comment]
@nikita1andreev
PLease provide me with simple indented stand alone code so i can replicate the issue.

**Comment by nikita1andreev** [comment]
Ok, here you are - BUG-REPORT.zip.
1. run, wait for fit, look at ETA - 2-3 min.
2. interrupt, run (model will be loaded from save), look at warnings, ETA - 32 min.
[BUG-REPORT.zip](https://github.com/tensorflow/tensorflow/files/4627829/BUG-REPORT.zip)

**Comment by gowthamkpr** [comment]
@nikita1andreev Tried to run this issue on colab and session is crashing

**Comment by nikita1andreev** [comment]
This version of ipynb needs for 4x smaller RAM, and i can run it in colab, but speed deceleration is not reproducted, it is about 40 sec from start, and same after loading from file.
I think the reason in differents between my local GPU - RTX2060, and colabs GPUs.
Another reason may be in smaller RAM for dataset (i dont think so), or some colab properties.

Try the original version on local environment with big RAM to reproduce bug. 

[BUG_REPORT.zip](https://github.com/tensorflow/tensorflow/files/4748993/BUG_REPORT.zip)

**Comment by Saduf2019** [comment]
@nikita1andreev 
Could you please try on tf 2.4.1 or later version and let us know if this is still an issue.
